skada/metrics.py

```diff
@@ -619,3 +619,131 @@ def _score(self, estimator, X, y, sample_domain=None):
+        Mixing parameter for mixup.
+    random_state : int, RandomState instance or None, default=None
+        Controls the randomness of the mixing process.
+    greater_is_better : bool, default=True
+        Whether higher scores are better.
+    ice_type : {'both', 'intra', 'inter'}, default='both'
+        Type of ICE score to compute:
+        - 'both': Compute both intra-cluster and inter-cluster ICE scores (average).
+        - 'intra': Compute only intra-cluster ICE score.
+        - 'inter': Compute only inter-cluster ICE score.
+
+    Attributes
+    ----------
+    alpha : float
+        Mixing parameter.
+    random_state : RandomState
+        Random number generator.
+    _sign : int
+        1 if greater_is_better is True, -1 otherwise.
+    ice_type : str
+        Type of ICE score to compute.
+
+    References
+    ----------
+    .. [32] Dapeng Hu et al. Mixed Samples as Probes for Unsupervised Model
+            Selection in Domain Adaptation.
+            NeurIPS, 2023.
+    """
+
+    def __init__(
+        self,
+        alpha=0.55,
+        random_state=None,
+        greater_is_better=True,
+        ice_type="both",
+    ):
+        super().__init__()
+        self.alpha = alpha
+        self.random_state = random_state
+        self._sign = 1 if greater_is_better else -1
+        self.ice_type = ice_type
+
+        if self.ice_type not in ["both", "intra", "inter"]:
+            raise ValueError("ice_type must be 'both', 'intra', or 'inter'")
+
+    def _score(self, estimator, X, y=None, sample_domain=None, **params):
+        """
+        Compute the Interpolation Consistency Evaluation (ICE) score.
+
+        Parameters
+        ----------
+        estimator : object
+            The fitted estimator to evaluate.
+        X : array-like of shape (n_samples, n_features)
+            The input samples.
+        y : Ignored
+            Not used, present for API consistency by convention.
+        sample_domain : array-like, default=None
+            Domain labels for each sample.
+
+        Returns
+        -------
+        score : float
+            The ICE score.
+        """
+        X, _, sample_domain = check_X_y_domain(X, y, sample_domain)
+        source_idx = extract_source_indices(sample_domain)
+        X_target = X[~source_idx]
+
+        rng = check_random_state(self.random_state)
+        rand_idx = rng.permutation(X_target.shape[0])
+
+        # Get predictions for target samples
+        pred_a = estimator.predict_proba(
+            X_target, sample_domain=sample_domain[~source_idx]
+        )
+        pl_a = pred_a.argmax(axis=1)
+        pl_b = pl_a[rand_idx]
+
+        # Intra-cluster and inter-cluster mixup
+        same_idx = (pl_a == pl_b).nonzero()[0]
+        diff_idx = (pl_a != pl_b).nonzero()[0]
+
+        # Mixup with images and hard pseudo labels
+        mix_inputs = self.alpha * X_target + (1 - self.alpha) * X_target[rand_idx]
+        mix_labels = pl_a if self.alpha > 0.5 else pl_b
```

Comment:
In the pseudo-code they are doing that, but neither in the paper nor in the GitHub they are using this formula for the mix_labels. They're using the formula with `alpha`, maybe it is the same ?

Reply:
Yes I agree, in the paper they're using the same formula as for X. However before going forward with the code, we need to agree what we should follow for all the implementation: The paper, the pseudo-code or the git code ?